spec/cordova/plugin/save.spec.js

```diff
@@ -54,15 +61,145 @@ describe('cordova/plugin/save', function () {
+            }).done(done);
+        });
+
+        it('should only add top-level plugins to config.xml', function (done) {
+            var fake_fetch_json =
+                {'VRPlugin': {'source': {
+                    'type': 'registry',
+                    'id': 'id'
+                },
+                'is_top_level': true
+                },
+                'MastodonSocialPlugin': { 'source': {
+                    'type': 'registry',
+                    'id': 'id'
+                },
+                'is_top_level': false }};
+
+            fs.readFileSync.and.returnValue(JSON.stringify(fake_fetch_json));
+            save(projectRoot).then(function () {
+                expect(cfg_parser_mock.prototype.addPlugin).toHaveBeenCalledWith(Object({ name: 'VRPlugin' }), [ ]);
+                expect(cfg_parser_mock.prototype.write).toHaveBeenCalled();
+            }).fail(function (e) {
+                expect(e).toBeUndefined();
+                fail('did not expect fail handler to be invoked');
+            }).done(done);
+        });
+
+        it('should write individual plugin specs to config.xml', function (done) {
+            var fake_fetch_json =
+                {'VRPlugin': {'source': {
+                    'type': 'registry',
+                    'id': 'id'
+                },
+                'is_top_level': true }};
+            fs.readFileSync.and.returnValue(JSON.stringify(fake_fetch_json));
+            spyOn(save, 'getSpec').and.returnValue('1.0.0');
+            save(projectRoot).then(function () {
+                expect(cfg_parser_mock.prototype.addPlugin).toHaveBeenCalledWith(Object({ name: 'VRPlugin', spec: '1.0.0' }), jasmine.any(Object));
+                expect(cfg_parser_mock.prototype.write).toHaveBeenCalled();
+            }).fail(function (e) {
+                expect(e).toBeUndefined();
+                fail('did not expect fail handler to be invoked');
+            }).done(done);
+        });
+
+        it('should write individual plugin variables to config.xml', function (done) {
+            var fake_fetch_json =
+                {'VRPlugin': {'source': {
+                    'type': 'registry',
+                    'id': 'id'
+                },
+                'is_top_level': true,
+                'variables': {
+                    'var 1': ' '
+                }}};
+            fs.readFileSync.and.returnValue(JSON.stringify(fake_fetch_json));
+            save(projectRoot).then(function () {
+                expect(cfg_parser_mock.prototype.addPlugin).toHaveBeenCalledWith(jasmine.any(Object), [ Object({ name: 'var 1', value: ' ' }) ]);
+                expect(cfg_parser_mock.prototype.write).toHaveBeenCalled();
+            }).fail(function (e) {
+                expect(e).toBeUndefined();
+                fail('did not expect fail handler to be invoked');
+            }).done(done);
+        });
     });
     describe('getSpec helper method', function () {
```

Comment:
So with all of these `getSpec` tests, you want to test getSpec directly. Right now, you are doing `save(projectRoot)` but instead you want to do `save.getSpec({ url: 'https://git-wip-us.apache.org/repos/asf/cordova-plugin-camera.git' }, '/some/path', 'VRPlugin')`